### PR TITLE
Add guidance around on-topic questions for Stack Overflow

### PR DESCRIPTION
**Pull request by sphanley**
This PR tweaks wording and adds guidance around the types of questions which are and are not appropriate to post on Stack Overflow. By directing users to the "How To Ask" guide before sending them to Stack Overflow, the likelihood of posting a well received, answerable question should be markedly improved.

Context for this change is described in #61.

### CODE REVIEW

**Comment by WillSullivan** [comment]
Just one note... the official spelling is "Stack Overflow". I prefer _StackOverflow_ myself, but the purists will puritan...

**Comment by sphanley** [comment]
Thanks Will - yeah, I corrected it to the correct spelling in one instance but missed another. I'm going to follow up on it.

File: contributor-guide/tools-and-setup.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 Follow the steps in this article to set up tools for contributing to the Bot Framework technical documentation. Casual and occasional contributors probably can use the GitHub UI described in step 2.
 
-If you're unfamiliar with Git, you might want to review some Git terminology: [https://help.github.com/articles/github-glossary](https://help.github.com/articles/github-glossary). In addition, this StackOverflow thread contains a glossary of Git terms you'll encounter here: [http://stackoverflow.com/questions/7076164/terminology-used-by-git](http://stackoverflow.com/questions/7076164/terminology-used-by-git)
```
Comment (grgarside) [inline contributor-guide/tools-and-setup.md]: Missing a space in Stack Overflow here

File: contributor-guide/tools-and-setup.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
...
+If you're unfamiliar with Git, you might want to review some Git terminology: [https://help.github.com/articles/github-glossary](https://help.github.com/articles/github-glossary). In addition, this StackOverflow post contains a glossary of Git terms you'll encounter here: [http://stackoverflow.com/questions/7076164/terminology-used-by-git](http://stackoverflow.com/questions/7076164/terminology-used-by-git)
```
Comment (grgarside) [inline contributor-guide/tools-and-setup.md]: Missing a space in Stack Overflow here

**Comment by RobStand** [comment]
@sphanley This is a great contribution. Thanks very much. Let me know when you are ready for me to merge.

**Comment by sphanley** [comment]
Thanks @RobStand! I pushed one more commit to fix the last "StackOverflow" instance, as was mentioned, so it should be ready to merge if you're happy with things!

**Comment by RobStand** [comment]
https://docs.microsoft.com/en-us/bot-framework/resources-links-help

**Comment by sphanley** [comment]
Great to see, @RobStand! Thanks again for being open to the changes! :smiley:

**Pull request merged**